# Phase 3j: equivalent forms of the call-only figures in the reports up to the end of 2024

Plan: `plans/phase_3j.md` (committed before the search, two amendments after). Computation: `analysis/phase3j_terms.py` (forms and terms), the check tool `data/phase3h/roman_check/check_terms.py` (search), `analysis/phase3j_read.py` and `analysis/phase3j_reading.py` (hits, passages, reading result); outputs in `data/phase3j/`. Descriptive: no test, no register entry.

In [1]:
from pathlib import Path
import json
import pandas as pd

ROOT = Path.cwd().parents[1] if Path.cwd().name == "roman" else Path.cwd()
OUT = ROOT / "notebooks" / "roman" / "data" / "phase3j"
pd.set_option("display.max_colwidth", 220, "display.max_rows", 80)

## Why

T32 was classed as mentioned without a figure, although the reports give the same increase in another unit (around USD 10bn of tier 1 capital against around 180 basis points). This phase searches the remaining eight units and T30 in every equivalent form of the call figure.

## Equivalent forms

In [2]:
forms = pd.read_csv(OUT / "forms.csv")
forms["searched values"] = forms.terms.str.split(" | ", regex=False).str.len()
forms[["unit", "form", "calculation", "call_figure", "central_value", "band_low", "band_high", "searched values"]]

,unit,form,calculation,call_figure,central_value,band_low,band_high,searched values
0,T33,amount,USD 2bn,2.0,2.00,1.80,2.20,4
1,T33,amount in CHF,2 / 1.19,2.0,1.68,1.51,1.85,2
2,T33,share of RWA,2 / 546.5,2.0,0.37,0.33,0.40,1
3,T33,share of RWA in basis points,2 / 546.5,2.0,36.60,32.94,40.26,6
4,T33,share of LRD,"2 / 1,695.4",2.0,0.12,0.11,0.13,1
5,T33,share of LRD in basis points,"2 / 1,695.4",2.0,11.80,10.62,12.98,3
6,T33,resulting AT1 stock,2.5% x 546.5 + 2,2.0,15.66,15.46,15.86,4
7,T33,resulting AT1 ratio,2.5% + 2 / 546.5,2.0,2.87,2.83,2.90,1
8,T68_r1,amount,USD 90bn,90.0,90.00,81.00,99.00,76
9,T68_r1,reduction,98 - 90,90.0,8.00,-1.00,17.00,68


Each form is a function of the call figure; its band is the image of the call figure plus and minus 10%. T68 is searched in two readings (`T68_r1`: gone concern capital, `T68_r2`: all funding raised through the holding company).

## Search

In [3]:
log = json.loads((OUT / "run_log.json").read_text())
pd.Series(log).to_frame("value")

,value
tool_sha256,c27228e8bee109eb636cdad7af4ae8cdc60273698e9fdb93ab6af726702f6e2d
check_units_sha256,1d64e39595ff62050cf8032d76e5c9c77eba774ab188bbe3b22898252cc99fa8
check_hits_sha256,dedec67364454a2260525e28db7566f8ba56272318657d81f4edb1db48c1a106
check_max_hits,100000
hits,12137
hits_to_read,2236
passages_to_read,874
distinct_passages,768
cap,600


In [4]:
summary = pd.read_csv(OUT / "summary.csv")
summary[["unit", "forms", "hits_pass_A", "hits_pass_B", "hits_pass_B_read", "passages_read", "distinct_passages_read"]]

,unit,forms,hits_pass_A,hits_pass_B,hits_pass_B_read,passages_read,distinct_passages_read
0,T33,8,18,974,70,43,43
1,T68_r1,5,343,2611,295,163,152
2,T68_r2,6,50,1831,135,103,81
3,F11,7,176,1637,146,134,124
4,F22,1,0,1156,301,82,82
5,T16,6,1,272,22,16,15
6,T25,4,0,246,4,3,3
7,T27,6,156,832,126,148,146
8,F12,5,127,1197,203,137,136
9,T30,4,13,497,50,45,45


Pass A (a figure form near a keyword) gives 904 hits, all read. Pass B (keywords alone) gives 11,233 hits, of which the 1,332 with a forward-looking word and a quantity in the context were read. Overlapping contexts form 874 passages; no unit exceeds the cap of 600.

## Classes of the passages read

In [5]:
classes = ["same target, same unit", "same target, another unit", "same target, another figure", "same target, no figure", "other forward-looking target", "result", "definition or requirement", "other"]
summary[["unit"] + classes]

,unit,"same target, same unit","same target, another unit","same target, another figure","same target, no figure",other forward-looking target,result,definition or requirement,other
0,T33,0,0,0,0,1,18,14,10
1,T68_r1,0,0,0,0,4,41,110,8
2,T68_r2,0,0,0,0,2,23,60,18
3,F11,0,0,0,0,1,45,64,24
4,F22,0,0,0,0,1,81,0,0
5,T16,0,0,0,0,2,7,7,0
6,T25,0,0,0,3,0,0,0,0
7,T27,0,0,0,3,29,97,0,19
8,F12,0,0,0,4,11,69,53,0
9,T30,0,1,0,0,9,34,0,1


All 768 distinct passages were read in full and judged one by one (`judgments_full.csv`). No passage states a call figure in the unit of the call, and none states another figure for the same target. One passage, for T30, gives the target in an equivalent form. Ten passages state the target of their unit without a figure.

## The one equivalent form: T30

In [6]:
read = pd.read_csv(OUT / "read.csv", keep_default_na=False)
read[read["class"].str.startswith("same target")][["unit", "file", "page", "class", "quote", "conversion", "comment"]]

,unit,file,page,class,quote,conversion,comment
541,T25,UBS_2023_annual_report.htm,56,"same target, no figure",freeing up capital for the UBS Group,,"The NCL key priority: the capital release as an aim, without a figure (class B counterpart of phase 3d)."
542,T25,UBS_2023_annual_report.htm,62,"same target, no figure",We plan to fund this growth organically from the capital released from the unwinding of the,,"The capital released by the NCL unwinding as a plan, without a figure (the phase 3d counterpart of T25)."
543,T25,UBS_2024_annual_report.htm,52,"same target, no figure",freeing up capital for the UBS Group,,"The NCL key priority: the capital release as an aim, without a figure (class B counterpart of phase 3d)."
561,T27,UBS_2023-Q3_report.htm,51,"same target, no figure",we expect exposures in Non-core and Legacy to reduce as a result of maturities and active unwinding of positions,,"Outlook of the credit risk RWA section: the reduction of NCL exposures as an expectation, without a figure; published before the call. T27 was already mentioned without a figure."
587,T27,UBS_2023_annual_report.htm,56,"same target, no figure","Reduce RWA and LRD, freeing up capital for the UBS Group",,"The NCL key priority: the RWA reduction as an aim, without a figure for credit and market risk RWA at the end of 2024; the target of around 5% of Group RWA in the same passage is for the end of 2026."
672,T27,UBS_2024_annual_report.htm,52,"same target, no figure",We will continue to actively wind down Non-core and Legacy,,"The continued wind-down as a plan, without a figure; published after the deadline of T27."
696,F12,UBS_2023-Q2_report.htm,8,"same target, no figure",We intend to actively reduce the assets of the NCL business division,,"The NCL part of the LRD reduction as an intention, without a figure; published before the call."
738,F12,UBS_2023_annual_report.htm,56,"same target, no figure","Reduce RWA and LRD, freeing up capital for the UBS Group",,"The NCL key priority: the LRD reduction as an aim, without a figure and for NCL only (class B counterpart of phase 3g)."
821,F12,UBS_2024_annual_report.htm,52,"same target, no figure",We will continue to actively wind down Non-core and Legacy,,"The continued wind-down of NCL as a plan, without a figure for the LRD."
822,F12,UBS_2024_annual_report.htm,52,"same target, no figure","Reduce RWA and LRD, freeing up capital for the UBS Group",,"The NCL key priority: the LRD reduction as an aim, without a figure and for NCL only (class B counterpart of phase 3g)."


The 2023-Q2 report expects the cumulative integration-related expenses to be broadly offset by accretion effects of approximately USD 12bn. This is within 10% of the call figure of around 13bn, but indirect, published before the call figure and already recorded in phase 3b. T30 is counted separately from the call-only figures.

## The same target without a figure

In [7]:
nf = read[read["class"] == "same target, no figure"]
nf[["unit", "file", "page", "quote", "comment"]]

,unit,file,page,quote,comment
541,T25,UBS_2023_annual_report.htm,56,freeing up capital for the UBS Group,"The NCL key priority: the capital release as an aim, without a figure (class B counterpart of phase 3d)."
542,T25,UBS_2023_annual_report.htm,62,We plan to fund this growth organically from the capital released from the unwinding of the,"The capital released by the NCL unwinding as a plan, without a figure (the phase 3d counterpart of T25)."
543,T25,UBS_2024_annual_report.htm,52,freeing up capital for the UBS Group,"The NCL key priority: the capital release as an aim, without a figure (class B counterpart of phase 3d)."
561,T27,UBS_2023-Q3_report.htm,51,we expect exposures in Non-core and Legacy to reduce as a result of maturities and active unwinding of positions,"Outlook of the credit risk RWA section: the reduction of NCL exposures as an expectation, without a figure; published before the call. T27 was already mentioned without a figure."
587,T27,UBS_2023_annual_report.htm,56,"Reduce RWA and LRD, freeing up capital for the UBS Group","The NCL key priority: the RWA reduction as an aim, without a figure for credit and market risk RWA at the end of 2024; the target of around 5% of Group RWA in the same passage is for the end of 2026."
672,T27,UBS_2024_annual_report.htm,52,We will continue to actively wind down Non-core and Legacy,"The continued wind-down as a plan, without a figure; published after the deadline of T27."
696,F12,UBS_2023-Q2_report.htm,8,We intend to actively reduce the assets of the NCL business division,"The NCL part of the LRD reduction as an intention, without a figure; published before the call."
738,F12,UBS_2023_annual_report.htm,56,"Reduce RWA and LRD, freeing up capital for the UBS Group","The NCL key priority: the LRD reduction as an aim, without a figure and for NCL only (class B counterpart of phase 3g)."
821,F12,UBS_2024_annual_report.htm,52,We will continue to actively wind down Non-core and Legacy,"The continued wind-down of NCL as a plan, without a figure for the LRD."
822,F12,UBS_2024_annual_report.htm,52,"Reduce RWA and LRD, freeing up capital for the UBS Group","The NCL key priority: the LRD reduction as an aim, without a figure and for NCL only (class B counterpart of phase 3g)."


These ten passages concern T25, T27 and F12, the three units that were already classed as mentioned without a figure. They add two earlier and two later sentences to the known counterparts and change no status.

## Other forward-looking targets

In [8]:
of = read[read["class"] == "other forward-looking target"]
of.groupby("unit").size().rename("passages").to_frame().T

unit,F11,F12,F22,T16,T27,T30,T33,T68_r1,T68_r2
passages,1,11,1,2,29,9,1,4,2


These passages state a target, plan or expectation on another quantity, scope or period: the NCL targets for the end of 2025 and 2026 (T27 has its deadline at the end of 2024), the quarterly amounts of integration-related expenses, the effect of the final Basel III standards, cost savings and the higher capital requirements.

## First reading against the complete reading

In [9]:
cmp = pd.read_csv(OUT / "reading_comparison.csv")
pd.crosstab(cmp.class_first, cmp.class_full, margins=True)

class_full,definition or requirement,other,other forward-looking target,result,"same target, another unit","same target, no figure",All
class_first,,,,,,,
definition or requirement,195,17,9,20,0,0,241
other,11,41,37,26,0,9,124
result,64,16,14,366,0,1,461
"same target, another unit",0,0,0,0,1,0,1
All,270,74,60,412,1,10,827


In [10]:
counted = ["same target, same unit", "same target, another unit", "same target, another figure"]
print("pairs of unit and passage:", len(cmp), "| class changed:", int(cmp.changed.sum()))
print("changes into or out of a class that counts:", int((cmp.class_first.isin(counted) != cmp.class_full.isin(counted)).sum()))

pairs of unit and passage: 827 | class changed: 224
changes into or out of a class that counts: 0


The first reading was sentence by sentence and sorted the passages that do not count by a wording rule. The complete reading changes the class of 224 of 827 pairs, all of them among the classes that do not count; the one equivalent form (T30) is the same in both readings.

## Hits per form in pass A

In [11]:
hits = pd.read_csv(OUT / "hits.csv", keep_default_na=False)
a = hits[hits.search_pass == "A"]
a.groupby(["unit", "form"]).size().rename("hits").reset_index()

,unit,form,hits
0,F11,AT1 capital,7
1,F11,AT1 increase,66
2,F11,AT1 ratio,53
3,F11,going concern capital,6
4,F11,increase in basis points,19
5,F11,ratio,17
6,F11,share of LRD,8
7,F12,CET1 equivalent at 3.5%,6
8,F12,amount,43
9,F12,capital equivalent at 5.0%,7


Most pass A hits come from small percentages and amounts that occur near a keyword by chance, and from reported values: the going concern and leverage ratios, the currency sensitivity of the LRD (around USD 100bn for a 10% move of the US dollar) and the LRD level of USD 1,599.6bn at 31 March 2024, which is a result.

## Result per unit

In [12]:
status = {"T33": "not mentioned", "T68_r1": "not mentioned", "T68_r2": "not mentioned", "F11": "not mentioned", "F22": "not mentioned", "T16": "mentioned without a figure", "T25": "mentioned without a figure", "T27": "mentioned without a figure", "F12": "mentioned without a figure", "T30": "counted separately"}
res = summary[["unit", "same target, same unit", "same target, another unit", "same target, another figure"]].copy()
res["status before"] = res.unit.map(status)
res["equivalent form found"] = (res.iloc[:, 1:4].sum(axis=1) > 0).map({True: "yes", False: "no"})
res

,unit,"same target, same unit","same target, another unit","same target, another figure",status before,equivalent form found
0,T33,0,0,0,not mentioned,no
1,T68_r1,0,0,0,not mentioned,no
2,T68_r2,0,0,0,not mentioned,no
3,F11,0,0,0,not mentioned,no
4,F22,0,0,0,not mentioned,no
5,T16,0,0,0,mentioned without a figure,no
6,T25,0,0,0,mentioned without a figure,no
7,T27,0,0,0,mentioned without a figure,no
8,F12,0,0,0,mentioned without a figure,no
9,T30,0,1,0,counted separately,yes


For the eight units T33, T68, F11, F22, T16, T25, T27 and F12, no report up to the end of 2024 states the target of the call in another unit or form. The count of call-only figures stays at 8, with T30 separate; the decision on T30 is Roman's.

## Limits

One reader. The forms are those listed in the plan; pass B guards against others but needs a keyword and a forward-looking word within 250 characters, so a target in a bare list would be missed. Not found means not found with these terms. The investor presentations were not searched.